clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -976,6 +976,24 @@ void CodeGenModule::Release() {
       Context.getTypeSizeInChars(Context.getWideCharType()).getQuantity();
   getModule().addModuleFlag(llvm::Module::Error, "wchar_size", WCharWidth);
 
+  if (getTriple().isOSzOS()) {
+    int32_t ProductVersion, ProductRelease, ProductPatch;
+    ProductVersion = LLVM_VERSION_MAJOR, ProductRelease = LLVM_VERSION_MINOR,
+    ProductPatch = LLVM_VERSION_PATCH;
+    getModule().addModuleFlag(llvm::Module::Warning, "Product Major Version",
+                              ProductVersion);
+    getModule().addModuleFlag(llvm::Module::Warning, "Product Minor Version",
+                              ProductRelease);
+    getModule().addModuleFlag(llvm::Module::Warning, "Product Patchlevel",
+                              ProductPatch);
```

Comment:
These are set only on z/OS, but the flag names appear generic.  If they're (set and) used only on z/OS, shouldn't they carry a z/OS specific name like below?

Reply:
This has been changed to use the `CLANG_VERSION_MAJOR`, `CLANG_VERSION_MINOR`, etc.